source/Ship.cpp

```diff
@@ -4048,6 +4048,27 @@ double Ship::BestFuel(const string &type, const string &subtype, double defaultF
 {
 	// Find the outfit that provides the least costly hyperjump.
 	double best = 0.;
+	double mass = Mass();
+
+	auto CalculateFuelCost = [this, mass, jumpDistance](const Outfit &outfit) -> double
+	{
+		// If the FTL drive has an attached "jump mass penalty"
+		// then the base fuel cost is assumed to be at 0 tons.
+		// Otherwise, base fuel cost is established to be at "jump mass".
+
+		// Temporary note: jump mass cost is fuel cost per 100 additional tons.
+
+		double fuel = outfit.Get("jump fuel");
+
+		// If any of these are not defined, they default to 0, so nothing will break.
+		fuel += .01 * outfit.Get("jump mass cost") * (mass - outfit.Get("jump base mass"));
+
+		// It is possible to generate a negative jump fuel, but I
+		// do not think this will mess with things in any major way,
+		// so I will not stop it.
+		return fuel;
+	};
```

Comment:
I would imagine most drives would have 0 mass = some nonzero fuel cost, rather than nonzero mass = zero fuel cost, but if you manage to get to zero fuel cost, I'd say shift the "default the fuel cost" value to -1, not 0, since that is an explicitly disallowed value

Reply:
That would require more changes farther down the line, which I'd rather not dip in to.